github/repos.go

```diff
@@ -1427,3 +1428,35 @@ func (s *RepositoriesService) Transfer(ctx context.Context, owner, repo string,
 
 	return r, resp, nil
 }
+
+// DispatchRequest represents a request to trigger a repository_dispatch event.
+type DispatchRequest struct {
+	// EventType is a custom webhook event name. (Required.)
+	EventType string `json:"event_type"`
+	// ClientPayload is a custom JSON payload with extra information about the webhook event.
+	// Defaults to an empty JSON object.
+	ClientPayload *json.RawMessage `json:"client_payload"`
```

Comment:
Are you sure this should be a pointer? Also, should this have `omitempty`?

This one field should have a bunch of table-driven unit tests to make sure that the payload is being passed and parsed correctly. Can you please add tests showing that this field works?

Reply:
1 and 2: These were addressed previously, but I forgot about the reason since last time, and un-addressed them in my last commit. I will re-address them again.
3: The reason it's a pointer is because it's optional, and I thought that was the best way to handle that. It definitely should have `omitempty`. And yes, it absolutely should have some unit tests. Do you have an example of a test in the code base where this is done in a good, idiomatic way?